azure-pipelines.yml

```diff
@@ -40,13 +40,19 @@ resources:
   - container: flink-build-container
     image: rmetzger/flink-ci:ubuntu-amd64-3528acd
 
-# See tools/azure-pipelines/jobs-template.yml for a short summary of the caching
+# Define variables:
+# - See tools/azure-pipelines/jobs-template.yml for a short summary of the caching
+# - See https://stackoverflow.com/questions/60742105/how-can-i-access-a-secret-value-from-an-azure-pipelines-expression
```

Comment:
Were you not trying `$[variables.IT_CASE_S3_BUCKET]` in your original post? what else had to be change for things to work?

Reply:
Currently on master, we have the `$(variables.IT_CASE_S3_BUCKET)` syntax, which, returns the variable, or `$(variables.IT_CASE_S3_BUCKET)` if nothing is set.
What I first proposed in the PR was `$[variables.IT_CASE_S3_BUCKET]`, which is always empty, if you use it in the `env` block of a task.
Now, I use this weird workaround of defining the variables in the pipeline definion.

I don't think this behavior is very intuitive.

https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch